openpgp-dsm/src/lib.rs

```diff
@@ -427,13 +490,24 @@ enum KeyRole {
     Subkey,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Default)]
 struct KeyMetadata {
-    certificate: String,
+    sq_dsm_version:              String,
+    fingerprint:                 String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    key_flags:                   Option<[u8; 2]>,
```

Comment:
Agreed on `Vec<u8>` instead of `[u8; 2]`. 
Good idea on `KeyFlags`. I'll remove everything, and for simplicity of implementation, just derive `Serialize, Deserialize` on `KeyFlags` and on `Bitfield`. This means that custom metadata will read something like e.g. `key_flags":{"raw":[12]}` which is fine.